Source/SwiftLintFramework/Protocols/SwiftSyntaxRule.swift

```diff
@@ -58,7 +58,53 @@ public extension SwiftSyntaxRule {
 }
 
 /// A SwiftSyntax `SyntaxVisitor` that produces absolute positions where violations should be reported.
-public protocol ViolationsSyntaxVisitor: SyntaxVisitor {
+public class ViolationsSyntaxVisitor: SyntaxVisitor {
```

Comment:
Let's make this `open` since it's designed to be subclassed.

```suggestion
open class ViolationsSyntaxVisitor: SyntaxVisitor {
```